subprojects/gst-plugins-bad/sys/msdk/gstmsdk.c

```diff
@@ -111,10 +116,194 @@ plugin_add_dependencies (GstPlugin * plugin)
+    case MFX_CODEC_MPEG2:
+      ret = gst_element_register (plugin, "msdkmpeg2enc", GST_RANK_NONE,
+          GST_TYPE_MSDKMPEG2ENC);
+      break;
+#ifdef USE_MSDK_VP9_ENC
+    case MFX_CODEC_VP9:
+      ret = gst_element_register (plugin, "msdkvp9enc", GST_RANK_NONE,
+         GST_TYPE_MSDKVP9ENC);
+      break;
+#endif
+#ifdef USE_MSDK_AV1_ENC
+    case MFX_CODEC_AV1:
+      ret = gst_element_register (plugin, "msdkav1enc", GST_RANK_NONE,
+         GST_TYPE_MSDKAV1ENC);
+      break;
+#endif
+    case MFX_CODEC_JPEG:
+     ret = gst_element_register (plugin, "msdkmjpegenc", GST_RANK_NONE,
+         GST_TYPE_MSDKMJPEGENC);
+      break;
+    default:
+      ret = FALSE;
+      break;
+  }
+
+  return ret;
+}
+
+static gboolean
+_register_decoder (GstPlugin * plugin, guint codec_id)
+{
+  gboolean ret = TRUE;
+
+  switch (codec_id) {
+    case MFX_CODEC_AVC:
+      ret = gst_element_register (plugin, "msdkh264dec", GST_RANK_NONE,
+          GST_TYPE_MSDKH264DEC);
+      break;
+    case MFX_CODEC_HEVC:
+      ret = gst_element_register (plugin, "msdkh265dec", GST_RANK_NONE,
+         GST_TYPE_MSDKH265DEC);
+      break;
+    case MFX_CODEC_MPEG2:
+      ret = gst_element_register (plugin, "msdkmpeg2dec", GST_RANK_NONE,
+         GST_TYPE_MSDKMPEG2DEC);
+      break;
+    case MFX_CODEC_VP8:
+      ret = gst_element_register (plugin, "msdkvp8dec", GST_RANK_NONE,
+        GST_TYPE_MSDKVP8DEC);
+      break;
+#ifdef USE_MSDK_VP9_DEC
+    case MFX_CODEC_VP9:
+      ret = gst_element_register (plugin, "msdkvp9dec", GST_RANK_NONE,
+         GST_TYPE_MSDKVP9DEC);
+      break;
+#endif
+#ifdef USE_MSDK_AV1_DEC
+    case MFX_CODEC_AV1:
+      ret = gst_element_register (plugin, "msdkav1dec", GST_RANK_NONE,
+          GST_TYPE_MSDKAV1DEC);
+      break;
+#endif
+    case MFX_CODEC_JPEG:
+      ret = gst_element_register (plugin, "msdkmjpegdec", GST_RANK_NONE,
+         GST_TYPE_MSDKMJPEGDEC);
+      break;
+    case MFX_CODEC_VC1:
+      ret = gst_element_register (plugin, "msdkvc1dec", GST_RANK_NONE,
+          GST_TYPE_MSDKVC1DEC);
+      break;
+    default:
+      ret = FALSE;
+      break;
+  }
+
+  return ret;
+}
+
+
+#if (MFX_VERSION >= 2000)
+
+static void
+_register_encoders (GstPlugin * plugin,
+    MsdkSession * session, mfxEncoderDescription * enc_desc)
+{
+  for (guint c = 0; c < enc_desc->NumCodecs; c++) {
+    if (!_register_encoder (plugin, enc_desc->Codecs[c].CodecID)) {
+      GST_WARNING ("Failed to register %"GST_FOURCC_FORMAT"ENC",
+          GST_FOURCC_ARGS (enc_desc->Codecs[c].CodecID));
+    }
+  }
+}
+
+static void
+_register_decoders (GstPlugin * plugin, mfxDecoderDescription * dec_desc)
+{
+  for (guint c = 0; c < dec_desc->NumCodecs; c++) {
+    if (!_register_decoder (plugin, dec_desc->Codecs[c].CodecID)) {
+      GST_WARNING ("Failed to register %"GST_FOURCC_FORMAT"DEC",
+          GST_FOURCC_ARGS (dec_desc->Codecs[c].CodecID));
+    }
+  }
+}
+
+static void
+_register_vpp (GstPlugin * plugin, mfxVPPDescription * vpp_desc)
```

Comment:
It looks like vpp_desc is not used in gst_element_register, or do you plan to use it in the future? If not, we may just have one _register_vpp for msdk and vpl.

Reply:
Yes, the API with the parameter mfxVPPDescription will be used after the dynamic caps is implemented in msdkvpp, and it will only be used in the VPL path, so the parameters of the two APIs are different.

Reply:
No need to fix.